elements/pl-variable-output/pl-variable-output.py

```diff
@@ -0,0 +1,126 @@
+
+            # Get value of variable, raising exception if variable does not exist
+            var_data = data['params'].get(var_name, None)
+            if var_data is None:
+                raise Exception('No value in data["params"] for variable %s in pl-variable-output element' % var_name)
+
+            # If the variable is in a format generated by pl.to_json, convert it
+            # back to a standard type (otherwise, do nothing)
+            var_data = pl.from_json(var_data)
+
+            # Get comment, if it exists
+            var_matlab_comment = ''
+            var_mathematica_comment = ''
+            var_python_comment = ''
+            if pl.has_attrib(child, 'comment'):
+                var_comment = pl.get_string_attrib(child, 'comment')
+                var_matlab_comment = '% {}'.format(var_comment)
+                var_mathematica_comment = '(* {} *)'.format(var_comment)
+                var_python_comment = '# {}'.format(var_comment)
+
+            # Get digit for child, if it exists
+            if not pl.has_attrib(child, 'digits'):
+                var_digits = digits
+            else:
+                var_digits = pl.get_string_attrib(child, 'digits')
+
+            # Assembling Python array formatting
+            if np.isscalar(var_data):
+                prefix = ''
+                suffix = ''
+            else:
+                # Wrap the variable in an ndarray (if it's already one, this does nothing)
+                var_data = np.array(var_data)
+                # Check shape of variable
+                if var_data.ndim > 2:
+                    raise Exception('Value in data["params"] for variable %s in pl-variable-output element must be a scalar, a vector, or a 2D array' % var_name)
+                # Create prefix/suffix so python string is np.array( ... )
+                prefix = 'np.array('
+                suffix = ')'
+
+            # Mathematica reserved letters: C D E I K N O
+            mathematica_reserved = ['C', 'D', 'E', 'I', 'K', 'N', 'O']
+            if pl.inner_html(child) in mathematica_reserved:
+                mathematica_suffix = 'm'
+            else:
+                mathematica_suffix = ''
+
+            # Create string for matlab and python format
+            var_name_disp = pl.inner_html(child)
+            var_matlab_data = pl.string_from_2darray(var_data, language='matlab', digits=var_digits)
+            var_mathematica = pl.string_from_2darray(var_data, language='mathematica', digits=var_digits)
+            var_python_data = pl.string_from_2darray(var_data, language='python', digits=var_digits)
+
+            matlab_data += f'{var_name_disp} = {var_matlab_data}; {var_matlab_comment}\n'
+            mathematica_data += f'{var_name_disp}{mathematica_suffix} = {var_mathematica}; {var_mathematica_comment}\n'
+            python_data += f'{var_name_disp} = {prefix}{var_python_data}{suffix} {var_python_comment}\n'
```

Comment:
It's a really minor point, but the current comment variable system will produce a trailing space if there is no comment. To avoid this, we could make the `var_matlab_comment` variable include a space at the start of it (if it's not the empty string) and change the template to end with `{var_matlab_data};{var_matlab_comment}` (no space between the semicolon and the comment variable.